### PR TITLE
January 2019 Update 

### LINKED ISSUE #59: Course Members page needs a redesign
The members page should either be:

Completely separate from /course/ (i.e. not in a tabbed box element)

or

Member info should come up as a modal box that can be dismissed.

It's annoying to look through member profiles because when you press back after looking at a profile, it reloads /course/ and you're taken to the course projects list.

### LINKED ISSUE #80: When professor creates a project, he is the TA and Scrum Master of the project.
can we set these fields to a null entity? Options...

### LINKED ISSUE #106: Need to apply User Role changes to Course view as well.
Currently edit_course throws error for TA's.

### LINKED ISSUE #109: TA's should be able to post Course Updates

### LINKED ISSUE #125: TSR Edit - Cancel Button
TSR Edit needs a cancel button. Also needs to be styled.

### LINKED ISSUE #148: Separate Instructors and TAs in Course Members tab
image

### LINKED ISSUE #149: New Feature: Disable ability to Request to Join project
Allow course owners to enable/disable students having the ability to request to join projects.

### LINKED ISSUE #156: Match Stats page doesn't have Breadcrumbs
Has 'Home' breadcrumb and thats it. Should be:

Home -> My Matches -> {Project Name} Match Stats

### LINKED ISSUE #157: Adjust notification dropdown
Currently the title 'Your Notifications' and your actual notifications look exactly the same.

image

### LINKED ISSUE #158: Adjust Account/Profile dropdown
Still doesn't look that great.

### LINKED ISSUE #169: Edit Resource
Add ability to edit a project resource

### PR DESCRIPTION
**Pull request by rmonroe**
**Tested**
 - Resolve #109 TA can't post course update
- Markdown Tooltips added to Bio and course description and changed to GitHub URL
- Multi-Course interest fix
- Change interest text to clarify Resubmit
- Resolve #158 Account drop-down rework 
- Resolve #59 and Resolve #148 Course member edit and split TAs and Profs
- Resolve #106 TA can edit Course
____________________________________________
**NEED TESTING**
- Added edit to project updates
- Fixes #156 Matchstats now has breadcrumbs
- Fixes #125 TSR cancel and re-design
- Professor can post resource for project
- Fixes #157 notification title fix
- Resolve #149 Add disable request to join
- Fixes #80 prof and ta no longer auto added as Scrum master and Ta
- Close #169 add ability to edit resource
- Add ability to delete a resource
- Resource no longer posted with UTC date
- Hides inactive dropdown if no inactive courses

### CODE REVIEW

File: teamwork/apps/courses/views/CourseView.py [hunk 1]
```diff
@@ -175,13 +179,18 @@ def update_course(request, slug):
...
+        user_role = 'GT'
+
     if request.user.profile.isGT:
         pass
-    #if user is not a professor or they did not create course
-    elif not course.creator == request.user:
-        #redirect them to the /course directory with message
-        messages.info(request,'Only Professor can post and update')
-        return HttpResponseRedirect('/course')
+    elif not request.user==course.creator:
```
Comment (Kapn) [inline teamwork/apps/courses/views/CourseView.py]: This entire section can just be:
``` 
if user_role == "student":
              create message
              redirect
```
We can delete the 4 extra lines of code. This is repeated in several views.

File: teamwork/apps/projects/migrations/0006_auto_20190125_2151.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
```
Comment (Kapn) [inline teamwork/apps/projects/migrations/0006_auto_20190125_2151.py]: Don't merge these into production until this PR gets the updates from the squash PR

File: teamwork/apps/projects/forms.py [hunk 3]
```diff
@@ -131,6 +131,10 @@ def __init__(self, uid, *args, **kwargs):
         max_length=20,
         required=False)
 
+    no_request = forms.BooleanField(
+        label="Do no allow Request to Join",
```
Comment (Kapn) [inline teamwork/apps/projects/forms.py]: Need to fix the label text. "Do no allow"

File: teamwork/templates/core/matchstats.html [hunk 4]
```diff
@@ -10,6 +10,10 @@
 
 {% endblock stylesheets %}
 
+{% block crumb %}
+<li><a href="{% url 'view_matches' %}">My Matches</a></li>
```
Comment (Kapn) [inline teamwork/templates/core/matchstats.html]: Breadcrumb for Matchstats should display: "Home > My Matches > Matchstats( with active css class)" currently doesn't show the current window.

File: teamwork/templates/core/view_matches.html [hunk 5]
```diff
@@ -10,9 +10,7 @@
   <link rel="stylesheet" href="{% static 'css/Tooltip.css' %}">
 {% endblock stylesheets %}
 
-{% block crumb %}
-<li><a href="{% url 'view_matches' %}">My Matches</a></li>
-{% endblock crumb %}
+
```
Comment (Kapn) [inline teamwork/templates/core/view_matches.html]: Breadcrumb for My Matches should display: "Home > My Matches (active)". Currently only displays "Home"

File: teamwork/templates/projects/add_resource.html [hunk 6]
```diff
@@ -2,14 +2,20 @@
 {% load static %}
 {% load i18n %}
 
+{% block crumb %}
```
Comment (Kapn) [inline teamwork/templates/projects/add_resource.html:5]: Same as others: Breadcrumb should be: "Home > CourseName > ProjectName > Add Resource (active)"

File: teamwork/templates/projects/edit_project.html [hunk 7]
```diff
@@ -305,6 +305,15 @@ <h3 class="box-title">Project Members</h3>
 <div class="col-md-6">
   <div class="box box-primary">
     <div class="box-header with-border">
+        <div class="form-group">
```
Comment (Kapn) [inline teamwork/templates/projects/edit_project.html]: Lock request to join shouldn't be in the same div as the box-title.

image

File: teamwork/templates/projects/post_update.html [hunk 8]
```diff
@@ -2,14 +2,20 @@
 {% load static %}
 {% load i18n %}
 
+{% block crumb %}
```
Comment (Kapn) [inline teamwork/templates/projects/post_update.html:5]: Same : "Home > Courses > CourseName > Post Update(active)

File: teamwork/templates/projects/tsr_member.html [hunk 9]
```diff
@@ -1,6 +1,15 @@
 {% extends 'base.html' %}
 {% load static %}
 {% load i18n %}
+
+{% block crumb %}
```
Comment (Kapn) [inline teamwork/templates/projects/tsr_member.html:5]: Same should be "Home > CourseName > ProjectName > TSR Edit

File: teamwork/templates/projects/tsr_scrum_master.html [hunk 10]
```diff
@@ -1,6 +1,15 @@
 {% extends 'base.html' %}
 {% load static %}
 {% load i18n %}
+
+{% block crumb %}
```
Comment (Kapn) [inline teamwork/templates/projects/tsr_scrum_master.html:5]: Same

File: teamwork/templates/projects/tsr_scrum_master.html [hunk 11]
```diff
@@ -28,20 +38,32 @@ <h3 class="box-title">Edit TSR Submission</h3>
...
-
                 {{ field }}
                 {% endfor %}
                 <br>
-                <br>
             {% endfor %}
-            <br>
-            <h5> <input type="submit" value="Submit">
+            <div class="form-group">
+                <div class="pull-left">
```
Comment (Kapn) [inline teamwork/templates/projects/tsr_scrum_master.html:57]: Form looks sweet

File: teamwork/templates/projects/update_post_update.html [hunk 12]
```diff
@@ -0,0 +1,81 @@
+{% extends 'base.html' %}
```
Comment (Kapn) [inline teamwork/templates/projects/update_post_update.html:1]: Looks good. We should add a message to display after editing/deleting an 'update'

File: teamwork/templates/projects/view_project.html [hunk 13]
```diff
@@ -229,7 +235,7 @@ <h3>Updates</h3>
                         </span>
                     </li>
                     <!-- /.timeline-label -->
-                    {% if user in project.members.all or user == project.creator or request.user.profile.isGT%}
+                    {% if user in project.members.all or user == course.creator or request.user.profile.isGT%}
```
Comment (Kapn) [inline teamwork/templates/projects/view_project.html]: Should we allow TA's to post project updates too?

**Review by Kapn — CHANGES_REQUESTED** [review]
Looks good, couple small tweaks needed in my opinion 💯

File: teamwork/apps/projects/migrations/0006_auto_20190125_2151.py (hunk 2 above)
Comment (rmonroe) [reply teamwork/apps/projects/migrations/0006_auto_20190125_2151.py]: Done

File: teamwork/templates/projects/tsr_scrum_master.html (hunk 11 above)
Comment (rmonroe) [reply teamwork/templates/projects/tsr_scrum_master.html:57]: Thanks

File: teamwork/apps/courses/views/CourseView.py (hunk 1 above)
Comment (rmonroe) [reply teamwork/apps/courses/views/CourseView.py]: Fixed

File: teamwork/templates/projects/update_post_update.html (hunk 12 above)
Comment (rmonroe) [reply teamwork/templates/projects/update_post_update.html:1]: Done for Resources as well

**Comment by rmonroe** [comment]
MERGE PR #211 FIRST, to add changes requested

**Comment by rmonroe** [comment]
Ensure you back up DB before, I had to run a DB flush after the migration Squash was merged in. 
Scrum master threw a NOT NULL constraint failed

**Review by Kapn — APPROVED** [review]

**Pull request merged**